src/hotspot/share/opto/loopnode.cpp

```diff
@@ -3997,7 +4031,7 @@ void PhaseIdealLoop::replace_parallel_iv(IdealLoopTree *loop) {
     // Instead we require 'stride_con2' to be a multiple of 'stride_con',
     // where +/-1 is the common case, but other integer multiples are
     // also easy to handle.
-    int ratio_con = stride_con2/stride_con;
+    jlong ratio_con = stride_con2 / stride_con;
 
     if ((ratio_con * stride_con) == stride_con2) { // Check for exact
```

Comment:
Was like that before but it might be easier to read when we negate this into an explicit skip. Otherwise, you first need to find the closing brace to ensure that we are not doing more things for the case when it's not equal:
```
if ((ratio_con * stride_con) != stride_con2) {
  continue;
}
```